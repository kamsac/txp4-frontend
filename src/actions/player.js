import dotenvConfiguration from '../dotenv-configuration';
import PlayerResourceMock from '../resources/player/mock';
import PlayerResource from '../resources/player';

export const RECEIVE_PLAYER = 'player:receivePlayer';
export function receivePlayer(player) {
  return {
    type: RECEIVE_PLAYER,
    payload: { player },
  };
}

function fetchPlayer() {
  const resource = dotenvConfiguration.API_URL ? PlayerResource : PlayerResourceMock;

  return (dispatch) => {
    resource.getPlayer()
      .then((response) => {
        dispatch(receivePlayer(response.data));
      });
  };
}

export const REQUEST_PLAYER = 'player:requestPlayer';
export function requestPlayer() {
  return (dispatch) => {
    dispatch(fetchPlayer());
  };
}
