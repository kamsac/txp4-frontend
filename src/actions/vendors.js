import VendorsResource from '../resources/vendors/mock';

export const REQUEST_VENDORS = 'vendors:requestVendors';
export function requestVendors() {
  return dispatch => {
    return dispatch(fetchVendors());
  }
}

function fetchVendors() {
  return dispatch => {
    VendorsResource.getVendors((vendors) => {
      dispatch(receiveVendors(vendors));
    });
  }
}

export const RECEIVE_VENDORS = 'vendors:fetchVendorsSuccess';
export function receiveVendors(vendors) {
  return {
    type: RECEIVE_VENDORS,
    vendors: vendors,
  }
}
