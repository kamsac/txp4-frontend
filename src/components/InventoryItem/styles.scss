@import '../../variables';

.InventoryItem {
  background-color: #ccc;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  border: $border-width solid #000;
  border-radius: $border-radius;
}

.InventoryItem--engine {
  background-image: url('../../images/items/engine-item-placeholder.png');
}

.InventoryItem--transmission {
  background-image: url('../../images/items/transmission-item-placeholder.png');
}

.InventoryItem--tire {
  background-image: url('../../images/items/tire-item-placeholder.png');
}

.InventoryItem--empty {
  background-color: #888;
}

.InventoryItem--selected {
  border-color: $primary-color;
}

.InventoryItem--vendor-accent {
  box-shadow: 10px 10px 10px 10px #f80;
}
