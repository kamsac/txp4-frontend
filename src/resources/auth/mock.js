export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
