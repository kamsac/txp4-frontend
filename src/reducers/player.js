import inventoryReducer from './inventory';
import { RECEIVE_PLAYER } from '../actions/player';
import { EQUIP_ITEM, RECEIVE_INVENTORY, REQUEST_INVENTORY } from '../actions/inventory';

const initialState = {
  login: '',
  nick: '',
  inventory: {
    items: [],
  },
};

export default function playerReducer(state = initialState, action) {
  switch (action.type) {
    case RECEIVE_PLAYER:
      return Object.assign({}, state, action.payload.player);

    case REQUEST_INVENTORY:
    case RECEIVE_INVENTORY:
    case EQUIP_ITEM: {
      const inventory = inventoryReducer(state.inventory, action);
      return Object.assign({}, state, { inventory });
    }
    default:
      return state;
  }
}
