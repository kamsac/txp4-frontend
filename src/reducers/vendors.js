import {RECEIVE_VENDORS} from '../actions/vendors';

const initialValue = [];

export function vendorsReducer(state = initialValue, action) {
  switch(action.type) {
    case RECEIVE_VENDORS:
      return [...action.vendors];
    default:
      return state;
  }
}
