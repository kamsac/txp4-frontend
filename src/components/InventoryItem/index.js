import React from 'react';
import './styles.scss';

export default class InventoryItem extends React.Component {
  constructor(props) {
    super(props);
  }

  handleClick(event) {
    if (this.props.onSelect) {
      this.props.onSelect(this.props.item);
    }
  }

  render() {
    const item = this.props.item;

    let className = 'InventoryItem';
    const imageClassName = `InventoryItem--${item.type}`;
    className += ` ${imageClassName}`;
    if (item === this.props.selectedItem) {
      className += ' InventoryItem--selected';
    }

    let styles = {};
    if (item.vendor) {
      styles.boxShadow = `inset 0 -4px 0 0 ${item.vendor.color}`;
    }

    return (
      <div onClick={(event) => this.handleClick(event)} className={className} style={styles} />
    );
  }
}
