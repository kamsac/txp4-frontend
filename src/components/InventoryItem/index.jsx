import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';
import { ItemPropTypesShape, VendorPropTypesShape } from '../../prop-types';

export default class InventoryItem extends React.Component {
  render() {
    const { item } = this.props;
    const vendor =
      this.props.vendors.find(iteratedVendor => iteratedVendor.id === this.props.item.vendorId);

    let className = 'InventoryItem';
    const imageClassName = `InventoryItem--${item.equipRegion}`;
    className += ` ${imageClassName}`;
    if (item === this.props.selectedItem) {
      className += ' InventoryItem--selected';
    }

    const styles = {};
    if (vendor) {
      styles.boxShadow = `inset 0 -4px 0 0 ${vendor.color}`;
    }

    return (
      <div
        className={className}
        style={styles}
        onClick={() => this.props.onSelectItem(this.props.item)}
        role="button"
        tabIndex={0}
        onKeyPress={() => null}
      />
    );
  }
}

InventoryItem.propTypes = {
  item: ItemPropTypesShape,
  vendors: PropTypes.arrayOf(VendorPropTypesShape).isRequired,
  onSelectItem: PropTypes.func,
  selectedItem: ItemPropTypesShape,
};

InventoryItem.defaultProps = {
  item: null,
  selectedItem: null,
  onSelectItem: () => null,
};
