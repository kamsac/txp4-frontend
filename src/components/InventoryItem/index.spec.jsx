import React from 'react';
import { shallow } from 'enzyme';
import InventoryItem from './index';
import { VENDORS } from '../../resources/vendors/mock';
import { ITEMS } from '../../resources/player/mock';

describe('<InventoryItem>', () => {
  test('with item', () => {
    const component = shallow(<InventoryItem
      item={ITEMS[0]}
      vendors={VENDORS}
      onSelectItem={jest.fn()}
      selectedItem={null}
    />);
    expect(component).toMatchSnapshot();
  });

  test('with no item', () => {
    const component = shallow(<InventoryItem
      item={null}
      vendors={VENDORS}
    />);
    expect(component).toMatchSnapshot();
  });

  it('should select item on click', () => {
    const onSelectItem = jest.fn();
    const component = shallow(<InventoryItem
      item={ITEMS[0]}
      vendors={VENDORS}
      onSelectItem={onSelectItem}
      selectedItem={null}
    />);
    component.simulate('click');
    expect(onSelectItem).toHaveBeenCalled();
  });
});
