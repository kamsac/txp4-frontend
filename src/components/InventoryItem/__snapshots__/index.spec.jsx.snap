// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InventoryItem> with item 1`] = `
<div
  className="InventoryItem InventoryItem--tire"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "boxShadow": "inset 0 -4px 0 0 #8f0",
    }
  }
  tabIndex={0}
/>
`;

exports[`<InventoryItem> with no item 1`] = `
<div
  className="InventoryItem"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={Object {}}
  tabIndex={0}
/>
`;
